spec/unit/outputs/elasticsearch/data_stream_support_spec.rb

```diff
@@ -0,0 +1,529 @@
+
+    it "does not use data-streams on LS 7.x" do
+      change_constant :LOGSTASH_VERSION, '7.10.0' do
+        expect( subject.data_stream_config? ).to be false
+      end
+    end
+
+    it "defaults to using data-streams on LS 8.0" do
+      change_constant :LOGSTASH_VERSION, '8.0.1' do
+        expect( subject.data_stream_config? ).to be true
+      end
+      change_constant :LOGSTASH_VERSION, '8.1.0' do
+        expect( subject.send(:check_data_stream_config!) ).to be true
+      end
+    end
+
+    # it "warns about not using data-streams on LS 8.0 (OSS)" do
+    #   expect( subject.logger ).to receive(:warn) do |msg|
+    #     expect(msg).to include "Configuration is data_stream compliant but won't be used"
+    #   end
+    #   change_constant :LOGSTASH_VERSION, '8.0.1' do
+    #     change_constant :OSS, true, target: LogStash do
+    #       expect( subject.data_stream_config? ).to be false
+    #     end
+    #   end
+    # end
+    it "uses data-streams on LS 8.0 (OSS)" do
+      change_constant :LOGSTASH_VERSION, '8.0.1' do
+        change_constant :OSS, true, target: LogStash do
+          expect( subject.data_stream_config? ).to be true
+        end
+      end
+    end
+
+    context 'old ES' do
+
+      let(:es_version) { '7.8.1' }
+
+      it "fails to start on LS 8.0" do
+        change_constant :LOGSTASH_VERSION, '8.0.0' do
+          change_constant :OSS, false, target: LogStash do
+            expect { subject.data_stream_config? }.to raise_error(LogStash::ConfigurationError, /data_stream is only supported since Elasticsearch/)
+          end
+        end
+      end
+
+      # it "uses index-ing on OSS LS 8.0" do
+      #   change_constant :LOGSTASH_VERSION, '8.0.0' do
+      #     change_constant :OSS, true, target: LogStash do
+      #       expect( subject.data_stream_config? ).to be false
+      #     end
+      #   end
+      # end
+      it "fails to start on OSS LS 8.0" do
+        change_constant :LOGSTASH_VERSION, '8.0.0' do
+          change_constant :OSS, true, target: LogStash do
+            expect { subject.data_stream_config? }.to raise_error(LogStash::ConfigurationError, /data_stream is only supported since Elasticsearch/)
+          end
+        end
+      end
+
+    end
+
+  end
+
+  context "default (non data-stream) configuration (on 7.x)" do
+
+    let(:options) do
+      { 'data_stream_dataset' => 'test', 'data_stream_auto_routing' => 'false', 'user' => 'elastic' }
+    end
+
+    before { allow(subject).to receive(:last_es_version).and_return(es_version) }
+
+    it "does not use data-streams by default" do
+      expect( subject.logger ).to receive(:warn) do |msg|
+        expect(msg).to include "Ambiguous configuration, data stream settings have no effect"
+      end
+      change_constant :LOGSTASH_VERSION, '7.10.2' do
+        expect { subject.data_stream_config? }.to raise_error(LogStash::ConfigurationError, /Ambiguous configuration/i)
```

Comment:
slightly more aggressive than specified initially - should be the best for the user.
on LS 7.x the default behavior is to use index-ing but given the user sets a data_stream_ param e.g.

`data_stream_dataset => test ...` in such cases we rather force to specify `data_stream => true/false`